File: app/src/main/java/com/example/aamezencev/weatherinfo/data/googleApi/JsonResultsGeo.java
package com.example.aamezencev.weatherinfo.data.googleApi;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by aa.mezencev on 12.01.2018.
 */

public class JsonResultsGeo {
    @SerializedName("results")
    private List<JsonReverseGeoModel> jsonReverseGeoModelList;

    public JsonResultsGeo(List<JsonReverseGeoModel> jsonReverseGeoModelList) {
        this.jsonReverseGeoModelList = jsonReverseGeoModelList;
    }

    public JsonResultsGeo() {

    }

    public List<JsonReverseGeoModel> getJsonReverseGeoModelList() {
        return jsonReverseGeoModelList;
    }

    public void setJsonReverseGeoModelList(List<JsonReverseGeoModel> jsonReverseGeoModelList) {
        this.jsonReverseGeoModelList = jsonReverseGeoModelList;
    }

    public JsonLocationModel getJsonLocationModel() {
        if (jsonReverseGeoModelList.size() != 0) {
            return jsonReverseGeoModelList.get(0).getJsonGeometryModel().getJsonLocationModel();
        }
        return null;
    }
}

File: app/src/main/java/com/example/aamezencev/weatherinfo/data/googleApi/JsonResultsGeo.kt
package com.example.aamezencev.weatherinfo.data.googleApi

import com.google.gson.annotations.SerializedName

/**
 * Created by aa.mezencev on 12.01.2018.
 */

class JsonResultsGeo {
    @SerializedName("results")
    var jsonReverseGeoModelList: List<JsonReverseGeoModel>? = null

    val jsonLocationModel: JsonLocationModel?
        get() = if (jsonReverseGeoModelList!!.size != 0) {
            jsonReverseGeoModelList!![0].jsonGeometryModel!!.jsonLocationModel
        } else null

    constructor(jsonReverseGeoModelList: List<JsonReverseGeoModel>) {
        this.jsonReverseGeoModelList = jsonReverseGeoModelList
    }

    constructor() {

    }
}
